src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -208,8 +214,10 @@ protected void executeDue() {
 			} else {
 				LOG.debug("Execution picked by another scheduler. Continuing to next due execution.");
 			}
-			count++;
+			pickedExecutionCount++;
 		}
+
+		lastPollAcquiredAllAvailableExecutors = pickedExecutionCount >= availableExecutorCount;
```

Comment:
Was looking at this line and was wondering if that is really needed, is it not sufficient to just check for `NoAvailableExecutors` exception?

Also, `pickedExecutionCount` will include executions that was picked by another Scheduler-instance, i.e. not running in this executor-pool

Reply:
Yeah, you are correct; Checking for `NoAvailableExecutors` should be enough.

My reason for including the additional logic was an edge case; But it should be quite rare so I will simplify the code 👍